articles/python-generators/index.md

```diff
@@ -0,0 +1,160 @@
+Generators are a special kind of function that return a lazy iterator, they were introduced with [PEP 255](https://www.python.org/dev/peps/pep-0255/). Generators are objects that you can loop through like a list.Lazy iterators do not store their contents in memory. The best way to iterate through large and complex datasets is through the use of generators.
+
+### Prerequisites
+To get started, you need to have a basic understanding of Python. All the functions and code used can be found on [repl.it](https://repl.it/@paulodhiambo962/PythonGenerators/). To execute various functions, click on **RUN**. You will now be able to access the variables and the function return values from the python interpreter. 
+
+**Generator function** is a function that returns a generator object. It is defined like a normal function, but whenever it needs to generate a value, it does so with the **yield** keyword rather than return. If the body of a def contains yield, the function automatically becomes a generator function.
```

Comment:
What do you mean by the body of def? Do you mean the body of function? It would be better to rephrase as: if the function contains the keyword yield, it automatically becomes a generator function.